// cezerin-client/lib/api/customerGroups.js
'use strict';

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var CustomerGroups = function () {
  function CustomerGroups(client) {
    _classCallCheck(this, CustomerGroups);

    this.client = client;
  }

  _createClass(CustomerGroups, [{
    key: 'list',
    value: function list(filter) {
      return this.client.get('/customer_groups', filter);
    }
  }, {
    key: 'retrieve',
    value: function retrieve(id, filter) {
      return this.client.get('/customer_groups/' + id, filter);
    }
  }, {
    key: 'create',
    value: function create(data) {
      return this.client.post('/customer_groups', data);
    }
  }, {
    key: 'update',
    value: function update(id, data) {
      return this.client.put('/customer_groups/' + id, data);
    }
  }, {
    key: 'delete',
    value: function _delete(id) {
      return this.client.delete('/customer_groups/' + id);
    }
  }]);

  return CustomerGroups;
}();

module.exports = CustomerGroups;
