// cezerin-client/lib/api/productCategories.js
'use strict';

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var ProductCategories = function () {
  function ProductCategories(client) {
    _classCallCheck(this, ProductCategories);

    this.client = client;
  }

  _createClass(ProductCategories, [{
    key: 'list',
    value: function list(filter) {
      return this.client.get('/product_categories', filter);
    }
  }, {
    key: 'retrieve',
    value: function retrieve(id) {
      return this.client.get('/product_categories/' + id);
    }
  }, {
    key: 'create',
    value: function create(data) {
      return this.client.post('/product_categories', data);
    }
  }, {
    key: 'update',
    value: function update(id, data) {
      return this.client.put('/product_categories/' + id, data);
    }
  }, {
    key: 'delete',
    value: function _delete(id) {
      return this.client.delete('/product_categories/' + id);
    }
  }, {
    key: 'uploadImage',
    value: function uploadImage(categoryId, formData) {
      return this.client.postFormData('/product_categories/' + categoryId + '/image', formData);
    }
  }, {
    key: 'deleteImage',
    value: function deleteImage(id) {
      return this.client.delete('/product_categories/' + id + '/image');
    }
  }]);

  return ProductCategories;
}();

module.exports = ProductCategories;
