// cezerin-client/lib/api/tokens.js
'use strict';

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var Tokens = function () {
  function Tokens(client) {
    _classCallCheck(this, Tokens);

    this.client = client;
  }

  _createClass(Tokens, [{
    key: 'list',
    value: function list(filter) {
      return this.client.get('/security/tokens', filter);
    }
  }, {
    key: 'retrieve',
    value: function retrieve(id) {
      return this.client.get('/security/tokens/' + id);
    }
  }, {
    key: 'create',
    value: function create(data) {
      return this.client.post('/security/tokens', data);
    }
  }, {
    key: 'update',
    value: function update(id, data) {
      return this.client.put('/security/tokens/' + id, data);
    }
  }, {
    key: 'delete',
    value: function _delete(id) {
      return this.client.delete('/security/tokens/' + id);
    }
  }]);

  return Tokens;
}();

module.exports = Tokens;
